uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/instances/leave-request.instance.spec.js

```diff
@@ -279,45 +279,101 @@ define([
-          instance.from_date_amount = 0;
-          instance.to_date_amount = 20;
-          promise = instance.calculateBalanceChange('hours');
+          calculateBalanceChange = instance.calculateBalanceChange('hours');
         });
 
-        afterEach(function () {
-          $rootScope.$apply();
-        });
+        describe('when set deductions are less than allowed', function () {
+          beforeEach(function () {
+            instance.from_date_amount = 1;
+            instance.to_date_amount = 1.5;
+            calculateBalanceChange.then(function (_balanceChange_) {
+              balanceChange = _balanceChange_;
+            });
+            $rootScope.$digest();
+          });
 
-        it('calls equivalent API method', function () {
-          promise.then(function () {
+          it('calls equivalent API method', function () {
             expect(LeaveRequestAPI.calculateBalanceChange)
               .toHaveBeenCalledWith(jasmine.objectContaining({
                 contact_id: jasmine.any(String),
                 from_date: jasmine.any(String),
                 to_date: jasmine.any(String)
               }));
           });
-        });
 
-        it('amends the breakdown "from" day amount', function () {
-          promise.then(function (balanceChange) {
+          it('amends the breakdown "from" day amount', function () {
             expect(_.first(balanceChange.breakdown).amount).toBe(instance.from_date_amount);
           });
-        });
 
-        it('amends the breakdown "to" day amount', function () {
-          promise.then(function (balanceChange) {
+          it('amends the breakdown "to" day amount', function () {
             expect(_.last(balanceChange.breakdown).amount).toBe(instance.to_date_amount);
           });
+
+          it('amends the breakdown', function () {
+            expect(balanceChange.amount).toBe(-5.5);
+          });
+        });
+
+        describe('when custom deductions are more than allowed', function () {
+          var workDayMaxPossibleDeductionFrom, workDayMaxPossibleDeductionTo;
+
+          beforeEach(function () {
+            var workDayFrom = leaveRequestMockData.workDayForDate().values;
+            var workDayTo = leaveRequestMockData.workDayForDate().values;
+
+            workDayMaxPossibleDeductionFrom =
+              moment.duration(workDayFrom.time_to)
+                .subtract(moment.duration(workDayFrom.time_from)).asHours();
+            workDayMaxPossibleDeductionTo =
+              moment.duration(workDayTo.time_to)
+                .subtract(moment.duration(workDayTo.time_from)).asHours();
+            instance.from_date_amount = 2000;
+            instance.to_date_amount = 1500;
+            calculateBalanceChange.then(function (_balanceChange_) {
+              balanceChange = _balanceChange_;
+            });
+            $rootScope.$digest();
+          });
+
+          it('reduces "from" day amount to the allowed value', function () {
```

Comment:
You tested what happens when both dates are outside of the new work pattern times, but where are the tests for when the from time is outside the range, but the to time is alright (and vice-versa)